### PR TITLE
Opening figures, simulink models, etc. in VS Code

### PR DESCRIPTION
**Pull request by Gusmano-2-OSU**
Implements opening `.fig`, `.slx` and other MATLAB files through VS code's context menu.

**Note!** I'm brand new to developing in TypeScript, and I'm having some implementation issues (specifically, getting the menu option to occur in the context menu.) Still, I wanted to put this here in case @dklilley or any of the other maintainers want to help with the final steps!

### CODE REVIEW

**Comment by dklilley** [comment]
Hi @Gusmano-2-OSU, thanks for working on this! I will plan to take a look at this soon (likely early next week).

Before I take a look, I just wanted to clarify - you are having some difficulty getting the "MATLAB: Open File" option added into the context menu?

**Comment by Gusmano-2-OSU** [comment]
Hey @dklilley, that's exactly right! The idea is, you right click any file supported by MATLAB's `open` command, and you get a context menu command that lets you run the command on that file. As someone who works a lot with figures and Simulink models, I'm hoping it can significantly increase the speed of my workflow!

My dream implementation of this is that a normal click would open these files as a tab in the current vs code window, but it would definitely be good enough if they opened in a separate MATLAB window.

Thanks for taking a look!

**Review by dklilley — CHANGES_REQUESTED** [review]

**Comment by dklilley** [comment]
I believe that the issue is just with where the configuration for the context menu item was being added within `package.json`. Take a look at my comment and see if it helps!

**Comment by dklilley** [comment]
I don't think there are any issues with the loop as you have written it. Instead, even if there are multiple files selected in the Explorer, only one URI is passed to the command.

Unfortunately, after some digging, I don't believe that this is functionality which is publicly available for extensions to leverage.

While there are some context menu actions which do work with multiple files selected (e.g. `Open to the Side`), these are commands provided my Microsoft and are implemented internally (e.g. the [implementation](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/files/browser/fileCommands.ts#L90-L118) of `Open to the Side` leverages an internal `getMultiSelectedResources` API).

**Comment by Gusmano-2-OSU** [comment]
That's too bad - maybe a project for another day if we could ever use the `getMultiSelectedResources` API. For now, I've reverted the commit and this should be ready to pull!

**Review by dklilley — APPROVED** [review]

**Comment by dklilley** [comment]
Agreed - hopefully they add some public functionality soon!

The changes look good to me! One of my colleagues will be reaching out soon with information about signing the CLA.

File: src/commandwindow/ExecutionCommandProvider.ts [hunk 1]
```diff
@@ -297,4 +297,29 @@ export default class ExecutionCommandProvider {
 
         void this._mvm.feval('cd', 0, [uri.fsPath]);
     }
+
+    /**
+     * Implements the open file action
+     * @param uri The file path to the file that should be opened
+     * @returns
+     */
+    async handleOpenFile(uri: vscode.Uri): Promise<void> {
```
Comment (dklilley) [inline src/commandwindow/ExecutionCommandProvider.ts:306]: Looks like a space is missing before the `(`, which is causing the linting to fail

Comment (Gusmano-2-OSU) [reply src/commandwindow/ExecutionCommandProvider.ts]: Should be taken care of!

Comment (dklilley) [reply src/commandwindow/ExecutionCommandProvider.ts]: Thanks!

**Comment by nothans** [comment]
Hi. Thanks for the contribution! Can you contact me about signing the CLA? I can help from there. Thanks.

**Comment by Gusmano-2-OSU** [comment]
I sent you an email, @nothans!

**Review by dklilley — APPROVED** [review]

**Comment by dklilley** [comment]
Thanks for working on this!

**Comment by Gusmano-2-OSU** [comment]
Sure thing! I look forward to contributing more in the future!

**Pull request merged**